### PR TITLE
Remove fire_event method usage

### PR DESCRIPTION
**Pull request by arku**
#### What? Why?

Closes https://github.com/openfoodfoundation/openfoodnetwork/issues/5324

- Removes the instrumentation of the following events:

  * `spree.checkout.update` - it does not have any consumers
  * `spree.order.contents_changed`- it has a consumer but it isn't necessary at all — the controller calls `update_distribution_charge` on the order
- Removes `fire_event` and  `default_notification_payload` methods
- Updates specs

#### What should we test?

From @luisramos0: We should test EnterpriseFees and verify if fees are calculated correctly on checkout and when creating orders in the back-office.

#### Release notes

Changelog Category: Added | Changed | Deprecated | Removed | Fixed | Security

### CODE REVIEW

**Review by luisramos0 — APPROVED** [review]
yeah, nice one, I think we can do this BUT I am afraid some edge case will not be covered and blow up... we need to take the risk and do the clean up anyway.
I think we should go ahead with this change.

**Comment by luisramos0** [comment]
Maybe we need to update "What to test" in the description:
I think we should test EnterpriseFees  and verify fees are correctly calculated on checkout and also when creating orders in the backoffice.

**Comment by arku** [comment]
@luisramos0 Thanks! I've updated the **What should we test?** section based on your comment.

**Comment by arku** [comment]
> I think we can do this BUT I am afraid some edge case will not be covered and blow up... we need to take the risk and do the clean up anyway.

@luisramos0 I have mentioned in the PR description why it's safe to remove the `fire_event` calls. Do let me know if I'm missing something.

**Comment by luisramos0** [comment]
no, your logic is correct :+1:

**Review by sauloperez — APPROVED** [review]

**Comment by sauloperez** [comment]
@arku because of the changes we are doing to the models, you'll need to rebase onto master and re-remove the contents_changed subscription.

**Comment by arku** [comment]
@sauloperez Done.

File: app/controllers/spree/orders_controller.rb [hunk 1]
```diff
@@ -78,11 +78,7 @@ def update
         discard_empty_line_items
         with_open_adjustments { update_totals_and_taxes }
 
-        if @order == current_order
-          fire_event('spree.order.contents_changed')
-        else
-          @order.update_distribution_charge!
-        end
```
Comment (luisramos0) [inline app/controllers/spree/orders_controller.rb:85]: we need to remove the consumer from app/models/spree/order.rb you
line 9:
ActiveSupport::Notifications...

Comment (arku) [reply app/controllers/spree/orders_controller.rb:85]: @luisramos0 Good catch! Removed.

**Review by sauloperez — APPROVED** [review]
I :heart: PRs removing code only :joy: . Thanks @arku !

**Review by luisramos0 — APPROVED** [review]

**Comment by luisramos0** [comment]
all right, this will be picked up by a testers soon and manually tested on a staging server.

**Comment by filipefurtad0** [comment]
Hey @arku ,

Thanks for another PR!

I did a sanity check on enterprise fees on this PR.

- created all types of fees, using the available calculators and added these to an open order cycle
- verified the fees included on each item (incl. price breakdown on the shopfront)
- placed different orders in the backoffice and shopfront, checking the values for each order total comparing the obtained values between backoffice and shopfront orders.
- verified the orders are displayed correctly on the relevant reports (/admin/reports/orders_and_distributors)

I found only matching values, the PR is good to go.

All the settings are documented here:
https://docs.google.com/document/d/10vZfj2ezuaz39RSmKdRBpumvbYBakTRUDcKuHzJDr20/edit#heading=h.9hcvpvb0ik9m

Thanks again Arun!


**Findings not introduced by this PR**

- rounding issues, related to #2522 :

image

and as reported [here](https://github.com/openfoodfoundation/openfoodnetwork/issues/2522#issuecomment-655461961). 

- rendering of the enterprise fees page:

image
image

I need to investigate, but I think this might be a recent regression.

**Comment by luisramos0** [comment]
that's weird Filipe, I dont think it's even related to 2522, there should be no rounding involved in the calculation... it's weird.

**Comment by filipefurtad0** [comment]
That's a good point Luis. I investigated this a bit more and I think this relates to the combination of:
- a `1 EUR per kg` fee
- a item of `1 g`.

Since we have a 1 EUR per kg fee:

image

If we introduce 10 Spearmint items in the cart (10 x 1 g = 10 g) we'll have a 1 cent fee here (0.01 EUR), which is correct. The example had 20 items, which led to a fee of 0.02 EUR.

What seems not to be correct, is the fact that this is only visible in the cart page. Visible on /cart on the order total (incorrect on the first line):

image

Not visible on the "side"-cart:

image

This is likely to be because the app is showing only two significant digits (0.01) in the price breakdown, but it seems to consider more, on following calculations:

image

But it becomes visible, if we have over 10 items in the cart.

Just to be sure, I reproduced this other staging server - all good with the PR:

image

**Comment by luisramos0** [comment]
ok Filipe, enterprise fees are not applicable to variants or line items, only to the full order. that could be playing a role.
I think weight based enterprise fees are not a real case, maybe we should switch to something more real, shipping methods? I wonder if this will also happen with a ship method. Anyway, I think this could deserve an issue. I think this is related but different from #2522 because it's about order level fees, not variant/line item prices.

**Comment by filipefurtad0** [comment]
Hi @luisramos0 ,

Thanks for your feedback, that's helpful. I agree - this deserves an issue. I've explored three different scenarios:

i) An enterprise fee of 1 EUR per kg 
ii) A shipping fee of 1 EUR per kg
iii) An enterprise fee of 0.1% per item

https://docs.google.com/document/d/1VU23pRPX-uOFcHlErCvc5SQxx-OT47vAHwkXBNlps8c/edit

The bug is only visible on scenario i), I've opened this issue #6260
